Index: configuration/ConfigInstrumentation.java
===================================================================
--- configuration/ConfigInstrumentation.java	(revision 8180)
+++ configuration/ConfigInstrumentation.java	(working copy)
@@ -36,6 +36,7 @@
     private List excludes;
 
     private List ignores;
+    private List ignoreMethods;
 
     private List includes;
 
@@ -47,6 +48,7 @@
         this.includes = new ArrayList();
         this.excludes = new ArrayList();
         this.ignores = new ArrayList();
+        this.ignoreMethods = new ArrayList();
         this.basedir = new File( System.getProperty( "user.dir" ) );
     }
 
@@ -71,6 +73,16 @@
     }
 
     /**
+     * Add an Ignored methods to the underlying list.
+     * 
+     * @param ignore the ignore string.
+     */
+    public void addIgnoreMethod( String ignore )
+    {
+        this.ignoreMethods.add( ignore );
+    }
+
+    /**
      * Add an Include ot the underlying list.
      * 
      * @param include the include string.
@@ -109,6 +121,16 @@
     }
 
     /**
+     * Get the Ignored method List.
+     * 
+     * @return the ignore list.
+     */
+    public List getIgnoreMethods()
+    {
+        return ignoreMethods;
+    }
+
+    /**
      * Get the Include List.
      * 
      * @return the include list.
@@ -187,6 +209,22 @@
             sb.append( "\"" );
         }
 
+        if ( !this.ignoreMethods.isEmpty() )
+        {
+            sb.append( " ignoreMethods=\"" );
+            Iterator it = this.ignoreMethods.iterator();
+            while ( it.hasNext() )
+            {
+                String ignore = (String) it.next();
+                sb.append( ignore );
+                if ( it.hasNext() )
+                {
+                    sb.append( " " );
+                }
+            }
+            sb.append( "\"" );
+        }
+
         return sb.append( " />" ).toString();
     }
 }
Index: tasks/InstrumentTask.java
===================================================================
--- tasks/InstrumentTask.java	(revision 8180)
+++ tasks/InstrumentTask.java	(working copy)
@@ -74,6 +74,13 @@
             cmdLineArgs.addArg( "--ignore", ignore );
         }
 
+        it = config.getIgnoreMethods().iterator();
+        while ( it.hasNext() )
+        {
+            String ignore = (String) it.next();
+            cmdLineArgs.addArg( "--ignoreMethod", ignore );
+        }
+
         String includes = joinCludes( config.getIncludes() );
         String excludes = joinCludes( config.getExcludes() );
         String defaultExcludes = joinCludes( FileUtils.getDefaultExcludesAsList() );
